sdk/react/src/lib/fragment-components/a-date-picker.tsx

```diff
@@ -0,0 +1,412 @@
+	nameStringToVariableString,
+	reactClassNamesFromComponentObj
+} from '../helpers/tools';
+import { css, cx } from 'emotion';
+
+const preventCheckEventStyle = css`
+	pointer-events: none;
+`;
+
+const checkboxStyle = css`
+	padding-top: 0.5rem;
+`;
+
+export const ADatePickerSettingsUI = ({ selectedComponent, setComponent }: any) => {
+	const [showRangePlaceholder, setShowRangePlaceholder] = useState(false);
+	const [rangePlaceholder, setRangePlaceholder] = useState('Range placeholder');
+
+	const size = [
+		{ id: 'sm', text: 'Small' },
+		{ id: 'md', text: 'Medium' },
+		{ id: 'lg', text: 'Large' }
+	];
+
+	const datePickerType = [
+		{ id: 'simple', text: 'Simple' },
+		{ id: 'single', text: 'Single with calender' },
+		{ id: 'range', text: 'Range with calender' }
+	];
+
+	return <>
+		<Checkbox
+			labelText='Light'
+			id='light'
+			checked={selectedComponent.light}
+			onChange={(_: any) => setComponent({
+				...selectedComponent,
+				light: !selectedComponent.light
+			})} />
+		<Checkbox
+			labelText='Disabled'
+			id='disable'
+			checked={selectedComponent.disabled}
+			onChange={(_: any) => setComponent({
+				...selectedComponent,
+				disabled: !selectedComponent.disabled
+			})} />
+		<Dropdown
+			id='datepicker-variation'
+			label='Date picker type'
+			titleText='Date picker type'
+			items={datePickerType}
+			selectedItem={datePickerType.find(item => item.id === selectedComponent.kind)}
+			itemToString={(item: any) => (item ? item.text : '')}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				kind: event.selectedItem.id,
+				...(event.selectedItem.id === 'range' && selectedComponent.rangeEndLabel === undefined) ? { rangeEndLabel: 'Range end' } : {}
+			})} />
+		{
+			(selectedComponent.kind === 'single' || selectedComponent.kind === 'range') &&
+			<TextInput
+				id='date-format'
+				value={selectedComponent.dateFormat}
+				labelText='Date format'
+				onChange={(event: any) => setComponent({
+					...selectedComponent,
+					dateFormat: event.currentTarget.value
+				})} />
+		}
+		<TextInput
+			id='placeholder'
+			value={selectedComponent.placeholder}
+			labelText='Placeholder'
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				placeholder: event.currentTarget.value
+			})} />
+		{selectedComponent.kind === 'range' &&
+			<Checkbox
+				className={checkboxStyle}
+				labelText='Override range placeholder'
+				id='override-range-placeholder'
+				checked={showRangePlaceholder}
+				onChange={(_: any, { checked }: any) => {
+					setShowRangePlaceholder(checked);
+					const componentAttributes = selectedComponent;
+
+					if (!checked) {
+						// Deleting the attributes from the model
+						delete componentAttributes.rangePlaceholder;
+					} else {
+						// Restore user set range placeholder
+						componentAttributes.rangePlaceholder = rangePlaceholder;
+					}
+
+					setComponent({
+						...componentAttributes
+					});
```

Comment:
There needs to be a better way to do what you're trying to do.

<img width="229" alt="image" src="https://github.com/user-attachments/assets/53d1d352-700a-4ee3-8ecb-7c776c10b806">